eng/pipelines/templates/jobs/archetype-sdk-tests-jobs.yml

```diff
@@ -0,0 +1,117 @@
+parameters:
+- name: Platforms
+  type: object
+  default: []
+- name: PreSteps
+  type: object
+  default: []
+- name: PostSteps
+  type: object
+  default: []
+- name: EnvVars
+  type: object
+  default: {}
+- name: MaxParallel
+  type: number
+  default: 0
+- name: BuildInParallel
+  type: boolean
+  default: true
+- name: TimeoutInMinutes
+  type: number
+  default: 60
+- name: Location
+  type: string
+  default: ''
+- name: SubscriptionConfiguration
+  type: string
+  default: $(sub-config-azure-cloud-test-resources)
+- name: ServiceDirectory
+  type: string
+  default: not-specified
+
+jobs:
+  - ${{ each platform in parameters.Platforms }}:
+    - job:
+      displayName: ${{ platform.DisplayName }}
+      variables:
+        - template: ../variables/globals.yml
+        - name: SubscriptionConfiguration
```

Comment:
Since we're no longer using `matrix` we don't need to set variables here, we can just use the template expansion. That is, delete the `SubscriptionConfiguration`, `Location`, and `TestTargetFramework` variables here and just replace the macro syntax expanded versions (e.g. `$(SubscriptionConfiguration)`) with the template syntax (e.g. `${{ platform.SubscriptionConfiguration }}`)... 

KEEP `OSVmImage` AS A VARIABLE IN THIS LIST FOR NOW. We have standardized on using that as a "global variable" in at least one place in the .NET code base and in other codebases. https://github.com/Azure/azure-sdk-for-net/blob/f642ecccb8af1cb22b24be2837c0255e7e077d6e/eng/common/pipelines/templates/steps/bypass-local-dns.yml#L6

Reply:
Fixed